source/slang/reflection.cpp

```diff
@@ -428,6 +428,47 @@ SLANG_API char const* spReflectionType_GetName(SlangReflectionType* inType)
     return nullptr;
 }
 
+SLANG_API SlangReflectionType * spReflection_FindTypeByName(SlangReflection * reflection, char const * name)
+{
+    auto context = convert(reflection);
+    auto compileRequest = context->targetRequest->compileRequest;
+
+    RefPtr<Type> result;
+    if (compileRequest->types.TryGetValue(name, result))
+        return (SlangReflectionType*)result.Ptr();
+
+    Decl* resultDecl = nullptr;
+    for (auto module : compileRequest->loadedModulesList)
```

Comment:
This loop will catch anything from `import`ed modules, but I think you also need to loop over the user's translation units in case they defined something in their own code rather than library code.